### ISSUE TITLE
Terminal:new does not source ~/.zshrc

### ISSUE THREAD

**Issue by uloco** [akinsho/toggleterm.nvim#317]
Is this by design? Because I am confused that the :ToggleTerminal commands just inherit my ~/.zshrc and Terminal:new does not. I was confused when setting up lazygit as described in your help file, because it would not find my custom config directory.

**Comment by akinsho** [comment]
@uloco so this is a bit of a misunderstanding about how this plugin works. It mainly opens a terminal with different types of windows. It does not control or manage how the terminal behaves, this is entirely up to neovim. Regarding using lazygit, presumably you pass the lazygit command directly to this plugin, which in turn passes it to neovim, and it controls what is loaded and what isn't.

When you run `ToggleTerm` it just opens a shell e.g. your shell so will ofc load your zshrc, when you run, `lazygit` it will just run that command on its own. If you read `:h termopen` and `:h jobstart` you will see that when passed a command directly it is run from the shell. As to if that is interactive and loads your zshrc this is definitely out of scope of this plugin.

I don't know where or how you have setup lazygit, but in my case, just putting the config in the correct place was enough for it to load my custom settings properly. This is AFAIK the same for other users. So, likely something is misconfigured on your side.

**Comment by uloco** [comment]
I have to use the lazygit function within my .zshrc because it takes into consideration what macos theme I have currently and has a different config directory etc. I did not know that there is a difference in opening a new terminal vs :toggleterm, this made things much clearer, thanks. From reading your help pages this was not clear though, because I have invested quite some time yesterday to figure this out. I just assumed it would use the same configuration as my interactive shell.
Thanks!
